doc/mutations/creating-mutation-plugins.md

```diff
@@ -1,7 +1,192 @@
+*  arguments - The arguments passed to the mutation. These are the fields for the entity you want to create, in the case above we are passing one argument called "Input" of type "ArticleInput". We will look at InputTypes afterwards. But essentially since graphql is strickly typed we want to provide information for types for each field we pass to the mutation we can do that using "InputTypes".
+
+### extractEntityInput method
+
+There is one method you should always implement when doing mutations, that is the **extractEntityInput** method which will be sort of a mapping between the arguments you pass to the mutation and the fields that drupal expects to receive for the entity being created.
+
+We can see we are assigning the **title** that we are passing in the input (we will look at the ArticleInput after) to the title property in the entity, same for the **body**.
+
+## Mutations to update Drupal Entities
+
+Lets continue with our article example. In this case we implement a mutation to update a given article. Because we are updating a particular entity and we need to know which entity it is, we will need to provide the plugin anotation with something extra, an Id for the entity.
+
+```
+<?php
+namespace Drupal\graphql_examples\Plugin\GraphQL\Mutations;
+use Drupal\graphql\Annotation\GraphQLMutation;
+use Drupal\graphql\GraphQL\Execution\ResolveContext;
+use Drupal\graphql_core\Plugin\GraphQL\Mutations\Entity\UpdateEntityBase;
+use GraphQL\Type\Definition\ResolveInfo;
+/**
+ * Simple mutation for updating an existing article node.
+ *
+ * @GraphQLMutation(
+ *   id = "update_article",
+ *   entity_type = "node",
+ *   entity_bundle = "article",
+ *   secure = true,
+ *   name = "updateArticle",
+ *   type = "EntityCrudOutput!",
+ *   arguments = {
+ *     "id" = "String",
+ *     "input" = "ArticleInput"
+ *   }
+ * )
+ */
+class UpdateArticle extends UpdateEntityBase {
+  /**
+   * {@inheritdoc}
+   */
+  protected function extractEntityInput(
+    $value,
+    array $args,
+    ResolveContext $context,
+    ResolveInfo $info
+  ) {
+    return array_filter([
+      'title' => $args['input']['title'],
+      'body' => $args['input']['body'],
+    ]);
+  }
+}
+```
+
+The first thing we noticed is we are now using **UpdateEntityBase** instead of "CreateEntityBase" as our parent class,
+we can also see that we use the same argument "Input" as above but we also have another argument called **id**. The Graphql Module will be smart enough to use that id to match to the right entity.
+
+## Mutations to Delete Drupal Entities
+
+The only thing left now is really to delete the entity right? This is the simples type of operation out of the 3, because we only need to give graphql the **id**, it will check if we can access that type of operation and if so delete the entity with the id we give to it. So lets look at how the plugin looks like 
```

Comment:
"The only thing left is to delete the entity, right?"

Reply:
FInish the sentence after "id". 

".. we only need to give the ``id``. It will check ..."